Terminal.Gui/Drawing/Color.Formatting.cs

```diff
@@ -0,0 +1,423 @@
+			int rangeCount = valuesSubstring.Split ( valueRanges, ',', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries );
+
+			switch ( rangeCount ) {
+			case 3:
+			{
+				// rgba(r,g,b)
+				ParseRgbValues ( in valuesSubstring, in valueRanges, in originalString, out ReadOnlySpan<char> rSpan, out ReadOnlySpan<char> gSpan, out ReadOnlySpan<char> bSpan );
+				return new Color ( int.Parse ( rSpan ), int.Parse ( gSpan ), int.Parse ( bSpan ) );
+			}
+			case 4:
+			{
+				// rgba(r,g,b,a)
+				ParseRgbValues ( in valuesSubstring, in valueRanges, in originalString, out ReadOnlySpan<char> rSpan, out ReadOnlySpan<char> gSpan, out ReadOnlySpan<char> bSpan );
+				ReadOnlySpan<char> aSpan = valuesSubstring [ valueRanges [ 3 ] ];
+				if ( !aSpan.IsAllAsciiDigits ( ) ) {
+					throw new ColorParseException ( in originalString, "Value was not composed entirely of decimal digits.", in aSpan, nameof ( A ) );
+				}
+				return new Color ( int.Parse ( rSpan ), int.Parse ( gSpan ), int.Parse ( bSpan ), int.Parse ( aSpan ) );
+			}
+			default:
+				throw new ColorParseException ( in originalString, $"Wrong number of values. Expected 3 or 4 decimal integers. Got {rangeCount}.", in originalString );
+			}
+
+			[Pure]
+			[SkipLocalsInit]
+			static void ParseRgbValues ( in ReadOnlySpan<char> valuesString, in Span<Range> valueComponentRanges, in ReadOnlySpan<char> originalString, out ReadOnlySpan<char> rSpan, out ReadOnlySpan<char> gSpan, out ReadOnlySpan<char> bSpan )
+			{
+
+				rSpan = valuesString [ valueComponentRanges [ 0 ] ];
+				if ( !rSpan.IsAllAsciiDigits ( ) ) {
+					throw new ColorParseException ( in originalString, "Value was not composed entirely of decimal digits.", in rSpan, nameof ( R ) );
+				}
+				gSpan = valuesString [ valueComponentRanges [ 1 ] ];
+				if ( !gSpan.IsAllAsciiDigits ( ) ) {
+					throw new ColorParseException ( in originalString, "Value was not composed entirely of decimal digits.", in gSpan, nameof ( G ) );
+				}
+				bSpan = valuesString [ valueComponentRanges [ 2 ] ];
+				if ( !bSpan.IsAllAsciiDigits ( ) ) {
+					throw new ColorParseException ( in originalString, "Value was not composed entirely of decimal digits.", in bSpan, nameof ( B ) );
+				}
+			}
+		}
+	}
+
+	/// <inheritdoc />
+	[Pure]
+	[SkipLocalsInit]
+	public static Color Parse ( ReadOnlySpan<byte> utf8Text, IFormatProvider? provider )
+	{
+		return Parse ( Encoding.UTF8.GetString ( utf8Text ), provider );
+	}
+
+	/// <inheritdoc cref="object.ToString" />
+	/// <summary>
+	///   Returns a <see langword="string" /> representation of the current <see cref="Color" /> value, according to the provided
+	///   <paramref name="formatString" /> and optional <paramref name="formatProvider" />.
+	/// </summary>
+	/// <param name="formatString">
+	///   A format string that will be passed to <see cref="string.Format(System.IFormatProvider?,string,object?[])" />. See remarks for parameters
+	///   passed to
```

Comment:
I'm a stickler for API doc stuff:

- Not sure this is a complete sentence.
- Sentences should end with periods (`.`).

Reply:
Darn it. I knew I was going to miss something when moving stuff around!

I'll take a pass over the XmlDoc to fix that and any other obvious things like that I see.

Reply:
And you know what....

Normally a plugin I use for resharper would have already told me about that, but apparently I never installed that plugin on this machine. It likes to enforce rules about grammar and general format of the content of XmlDoc comments, as well as content of logger output (though that's not something we have at this time and should consider at some point).

Reply:
Fixed in upcoming commit